scripts/pylib/twister/twisterlib.py

```diff
@@ -2017,6 +2017,7 @@ def run_cmake(self, args=[]):
         cmake_args = [
             f'-B{self.build_dir}',
             f'-S{self.source_dir}',
+            f'-DEMU_PLATFORM_TARGET={self.platform.simulation}',
```

Comment:
I don't think this is needed in twister.
```suggestion
```

Extra arguments to CMake can already be provided like this:
https://github.com/zephyrproject-rtos/zephyr/blob/3053a931a31b0f72a7b149edf8ed2a244f8273a9/tests/misc/test_build/testcase.yaml#L5

Reply:
Right. I've dropped this change.

Reply:
but twister will use the default board per default, which we defined to be the first board in the list.

Which you are doing here:
https://github.com/zephyrproject-rtos/zephyr/blob/7334d3504c1389a61f66ff40fbf12a4b1b66d994/boards/riscv/hifive1/board.cmake#L3

So what is the value in specifying the same emulation software here, as it will be the same as the default:
https://github.com/zephyrproject-rtos/zephyr/blob/7334d3504c1389a61f66ff40fbf12a4b1b66d994/boards/riscv/hifive1/hifive1.yaml#L8

Are there any usecases where the `simulation: <emulator>` in `<board-dir>/<board>.yaml` would be different than the default emulator in `<board-dir>/board.cmake` ?
I would assume we would like twister to use the default emulator, unless explicitly specified in a `testcase.yaml` file.